Yoast/YoastSEO.js
# Add Node inheritance to StructuredNode

## Summary

This PR can be summarized in the following changelog entry:

* [not-user-facing] Renames `StructuredText` to `StructuredNode`
* [not-user-facing] Adds `Node` inheritance to `StructuredNode`
* [not-user-facing] Adds a `tag` attribute to `StructuredNode`

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test`, check if all tests pass and that the coverage is 100%.

Fixes #2024
